### ISSUE TITLE
Integration container build errors are absent from the log messages

### ISSUE THREAD

**Issue by orpiske** [apache/camel-k#2306]
In some cases if the builder fails (i.e.: if it fails to pull the base image), the information about this failure is absent from the logs. This makes debugging build errors much more complicated.

Log snippet on a build that failed because it could not pull the base image:
```
{"level":"info","ts":1621320008.451126,"logger":"camel-k","msg":"spectrum - 2021/05/18 06:40:08 Pulling base image localhost:5000/openjdk11:slim (insecure=true)..."}
{"level":"info","ts":1621320008.4511755,"logger":"camel-k","msg":"spectrum - 2021/05/18 06:40:08 No matching credentials were found, falling back on anonymous"}
{"level":"info","ts":1621320008.6820612,"logger":"camel-k.controller.build","msg":"Build state transition","request-namespace":"default","request-name":"kit-c2hm1tlcbeqlprug5nn0","api-version":"camel.ap
ache.org/v1","kind":"Build","ns":"default","name":"kit-c2hm1tlcbeqlprug5nn0","phase":"Failed"}
 {"level":"info","ts":1621320008.6868963,"logger":"camel-k.controller.build","msg":"Reconciling Build","request-namespace":"default","request-name":"kit-c2hm1tlcbeqlprug5nn0"}
{"level":"info","ts":1621320008.687199,"logger":"camel-k.controller.integrationkit","msg":"Reconciling IntegrationKit","request-namespace":"default","request-name":"kit-c2hm1tlcbeqlprug5nn0"}
{"level":"info","ts":1621320008.6882634,"logger":"camel-k.controller.integrationkit","msg":"Invoking action build-submitted","request-namespace":"default","request-name":"kit-c2hm1tlcbeqlprug5nn0","api-
version":"camel.apache.org/v1","kind":"IntegrationKit","ns":"default","name":"kit-c2hm1tlcbeqlprug5nn0"}
{"level":"info","ts":1621320008.6882765,"logger":"camel-k.controller.build","msg":"Invoking action error-recovery","request-namespace":"default","request-name":"kit-c2hm1tlcbeqlprug5nn0","api-version":"
camel.apache.org/v1","kind":"Build","ns":"default","name":"kit-c2hm1tlcbeqlprug5nn0"}
```

**Comment by orpiske** [comment]
I'll send a PR for this one.

**Comment by astefanutti** [comment]
It should be fixed with #2269.
